rust-lite/src/rust_lite/cli.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+import logging
+from argparse import ArgumentParser
+from functools import cached_property
+from typing import TYPE_CHECKING, Any
+
+from pyk.cli.args import DisplayOptions as PykDisplayOptions
+from pyk.cli.args import KCLIArgs, KDefinitionOptions, LoggingOptions, Options, SaveDirOptions
+from pyk.cli.utils import file_path
+from pyk.kast.inner import KSort
+from pyk.kore.tools import PrintOutput
+from pyk.ktool.krun import KRunOutput
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from pathlib import Path
+    from typing import Final, TypeVar
+
+    T = TypeVar('T')
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+def generate_options(args: dict[str, Any]) -> LoggingOptions:
+    command = args['command']
+    match command:
+        case 'version':
+            return VersionOptions(args)
+        case 'run':
+            return RunOptions(args)
+        case _:
+            raise ValueError(f'Unrecognized command: {command}')
+
+
+def get_option_string_destination(command: str, option_string: str) -> str:
+    option_string_destinations = {}
+    match command:
+        case 'version':
+            option_string_destinations = VersionOptions.from_option_string()
+        case 'parse':
+            option_string_destinations = RunOptions.from_option_string()
+
+    return option_string_destinations.get(option_string, option_string.replace('-', '_'))
+
+
+def get_argument_type_setter(command: str, option_string: str) -> Callable[[str], Any]:
+    def func(par: str) -> str:
+        return par
+
+    option_types = {}
+    match command:
+        case 'version':
+            option_types = VersionOptions.get_argument_type()
+        case 'parse':
+            option_types = RunOptions.get_argument_type()
+
+    return option_types.get(option_string, func)
+
+
+def _create_argument_parser() -> ArgumentParser:
+    rust_lite_cli_args = RustLiteCLIArgs()
+    parser = ArgumentParser(prog='rust-lite')
+
+    command_parser = parser.add_subparsers(dest='command', required=True)
+
+    # command_parser.add_parser('version', help='Print Rust-Lite version and exit.', parents=[rust_lite_cli_args.logging_args])
```

Comment:
I think it's fine to leave these here if you plan to uncomment them soon, but otherwise you should consider deleting them.